src/libprojectM/CMakeLists.txt

```diff
@@ -0,0 +1,245 @@
+add_compile_definitions(
+        DATADIR_PATH="${PROJECTM_DATADIR_PATH}"
+        GL_SILENCE_DEPRECATION
+)
+
+add_subdirectory(MilkdropPresetFactory)
+add_subdirectory(NativePresetFactory)
+add_subdirectory(Renderer)
+
+# CMake cannot combine multiple static libraries using target_link_libraries.
+# This syntax will pull in the compiled object files into the final library.
+add_library(projectM_static STATIC
+        $<TARGET_OBJECTS:MilkdropPresetFactory>
+        $<TARGET_OBJECTS:NativePresetFactory>
+        $<TARGET_OBJECTS:Renderer>
+        $<TARGET_OBJECTS:hlslparser>
+        $<TARGET_OBJECTS:SOIL2>
+        Common.hpp
+        ConfigFile.cpp
+        ConfigFile.h
```

Comment:
Does adding header files do anything?

Reply:
Not required for compilation, but adds them to the "rebuild target and its dependents if these files have changed" list and also displays the files in IDE projects, e.g. Xcode, KDevelop or Visual Studio so one can easily open headers from the project tree. Also makes sure that the configuration phase fails if the file is not present for some reason.